core/src/main/java/org/frankframework/pipes/IfPipe.java

```diff
@@ -0,0 +1,407 @@
+				&& !jsonArray.isEmpty()) {
+			return jsonArray.get(0).toString();
+		}
+
+		return null;
+	}
+
+	private String getInputString(Message message) throws PipeRunException {
+		if (Message.isEmpty(message)) {
+			return "";
+		} else {
+			try {
+				return message.asString();
+			} catch (IOException e) {
+				throw new PipeRunException(this, "cannot open stream", e);
+			}
+		}
+	}
+
+	private PipeForward getForwardForStringInput(String inputString) {
+		if (StringUtils.isNotEmpty(regex)) {
+			return inputString.matches(regex) ? thenForward : elseForward;
+		} else if (StringUtils.isNotEmpty(expressionValue)) {
+			return inputString.equals(expressionValue) ? thenForward : elseForward;
+		}
+
+		// If the input is not empty, use then forward.
+		return StringUtils.isNotEmpty(inputString) ? thenForward : elseForward;
+	}
+
+	private PipeForward assertExistsAndGetForward(String forwardName) throws ConfigurationException {
+		PipeForward forward = findForward(forwardName);
+
+		if (forward != null) {
+			return forward;
+		}
+
+		throw new ConfigurationException("has no forward with name [" + forwardName + "]");
+	}
+
+	/**
+	 * forward returned when output is <code>true</code>
+	 *
+	 * @ff.default then
+	 */
+	@Deprecated(forRemoval = true, since = "9.0")
+	@ConfigurationWarning(value = "Use the 'then' forward in your configuration")
+	public void setThenForwardName(String thenForwardName) {
+		this.thenForwardName = thenForwardName;
+	}
+
+	/**
+	 * forward returned when output is <code>false</code>
+	 *
+	 * @ff.default else
+	 */
+	@Deprecated(forRemoval = true, since = "9.0")
+	@ConfigurationWarning(value = "Use the 'else' forward in your configuration")
+	public void setElseForwardName(String elseForwardName) {
+		this.elseForwardName = elseForwardName;
+	}
+
+	/** a string to compare the result of the xpathExpression (or the input-message itself) to. If not specified, a non-empty result leads to the 'then'-forward, an empty result to 'else'-forward */
+	public void setExpressionValue(String expressionValue) {
+		this.expressionValue = expressionValue;
+	}
+
+	/** xpath expression to be applied to the input-message. if not set, no transformation is done when the input message is mediatype XML */
+	public void setXpathExpression(String string) {
+		xpathExpression = string;
+	}
+
+	/** jsonPath expression to be applied to the input-message. if not set, no transformation is done when the input message is mediatype JSON */
+	public void setJsonPathExpression(String jsonPathExpression) {
+		this.jsonPathExpression = jsonPathExpression;
+	}
+
+	/**
+	 * Regular expression to be applied to the input-message (ignored if <code>xpathExpression</code> is specified).
+	 * The input-message <b>fully</b> matching the given regular expression leads to the 'then'-forward
+	 */
+	@Deprecated(forRemoval = true, since = "9.0")
+	@ConfigurationWarning(value = "Use RegExPipe instead")
+	public void setRegex(String regex) {
+		this.regex = regex;
+	}
+
```

Comment:
Ignored if either xPath or JSON path expression is configured.

I'm wondering though, this is a new pipe, right? Where does this "legacy" behaviour come from?

Also, this is not described in the Java/Frank!Doc at the top so it was a bit of a surprise to me to find it eventually.
Do we need to keep this feature in here, in this new pipe?

Reply:
Dit komt uit de xmlIf pipe, het idee hierbij is dat een gebruiker straks in principe de een voor de ander kan inwisselen met (idealiter) geen wijzigingen in de configuratie

Reply:
Maar tegelijkertijd willen we ook dat ze overschakelen naar de RegexPipe als ze een regex gebruiken. Ik weet niet of dat misschien voor verwarring gaat zorgen bij gebruikers.

Ik ben er niet van overtuigd dat dit de beste manier is, maar Ok.

Reply:
Helemaal mee eens hoor, mooi om te zien dat er zoveel discussie op gang komt door dit soort changes ;-) 

Als dit in 9.0 is deprecated, willen we het dan nu al niet meenemen naar de IfPipe?

Reply:
XmlIfPipe extends IfPipe (zonder regex) dan kan in de xml-if wel de regex blijven, en deze gehele pipe gedeprecate worden.

Reply:
Als `XmlIfPipe extends IfPipe`, is het dan mogelijk om in de `XmlIfPipe` het `jsonPathExpression` attribuut te verbergen? Het lijkt mij niet wenselijk als mensen dat gaan configureren op die pipe, in plaats van te upgraden naar de generieke `IfPipe`. En dan hoeven we het ook niet werkend te krijgen icm. de xpath en regex.
Dan hebben beide pipe betere code.

Reply:
Mee eens, goed punt. Ik begreep dat dit kan dmv @Protected in de XmlIf

Reply:
Ik had na deze discussie verwacht dat regex-features zouden worden weggehaald uit de `IfPipe` en alleen in de `XmlIfPipe` zouden blijven, maar ik kan het verkeerd begrepen hebben?

Reply:
@tnleeuw , ik ga het XmlIf extends IfPipe in een tweede branch verder uitwerken om dit niet te groot te maken